src/new_hope/week1.clj

```diff
@@ -40,13 +40,82 @@
   (second-to-last [1]) => nil
   (second-to-last nil) => nil)
 
-(comment
-  ;; Simple time measurement (run once)
-  (let [coll (vec (range 10000))]
-    (time (second (reverse coll)))
-    (time (last (butlast coll))))
-  ;; More elaborate benchmark taking 10 seconds
-  (let [coll (vec (range 10000))]
-    (crit/quick-bench (second (reverse coll)))
-    (crit/quick-bench (last (butlast coll))))
-  )
+
+
+;(defn sum-it-up
+;  [coll]
+;  (if (empty? coll)
+;    0
+;    (+
+;      (first coll)
+;      (summ-it-up (rest coll))
+;      )
+;    )
+;  )
+
+; And then I read about "reduce"
+(def sum-it-up #(reduce + %))
+
+(facts "about sum-it-up"
+       (sum-it-up [1 2 3]) => 6
+       (sum-it-up (list 0 -2 5 5)) => 8
+       (sum-it-up #{4 2 1}) => 7,
+       (sum-it-up '(0 0 -1)) => -1
+       (sum-it-up '(1 10 3)) => 14
+       ;Edge cases
+       (sum-it-up '(1)) => 1
+       (sum-it-up '()) => 0
+       )
+
+
+
+(def find-odd #(filter odd? %))
+
+(facts "about find-odd"
+       (find-odd #{1 2 3 4 5}) => '(1 3 5)
+       ;Edge cases
+       (find-odd '(1)) => '(1)
+       (find-odd '(2)) => '()
+       (find-odd '()) => '()
+       )
+
+
+
+(defn palindrom? [p]
+  (if (or (empty? p) (= 1 (count p)))
+      true
+      (if (not= (first p) (last p))
+        false
+;        (palindrom? (drop 1 (drop-last 1 p))))))
+        (recur (drop 1 (drop-last 1 p))))))
+
+(facts "about palindrom"
+       (palindrom? '(1 2 3 4 5)) => false
```

Comment:
Ok, but most test are copy-paste form book. IE: `(true? (__ '(:a :b :c)))`. They are using lists for this. *That one I will ask you on Monday.*